### PR TITLE
Deactivate jetpack widgets when VIP search is enabled

### PR DESCRIPTION
**Pull request by hanifn**
## Description

Deactivate all the automatically registered Jetpack widgets when VIP Search is enabled as the widgets will force the site to keep using Jetpack search.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.

## Steps to Test

1. Pull down PR
1. Enable JP Search + a widget on a test site
1. Set `USE_VIP_ELASTICSEARCH` to true in wp-config.php
1. Run a query on the frontend - JP Search should not enable itself (check Debug Bar for any Elasticsearch queries from JP)

### CODE REVIEW

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 1]
```diff
@@ -106,4 +109,8 @@ public function filter__jetpack_active_modules( $modules ) {
 		// array_filter() preserves keys, so to get a clean / flat array we must pass it through array_values()
 		return array_values( $filtered );
 	}
+
+	public function filter__jetpack_widgets_to_include( $widget_include ) {
+		return [];
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-elasticsearch.php]: This is going to disable _every_ JP widget - we only need to disable the Search one.

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 2]
```diff
@@ -54,6 +54,9 @@ protected function setup_hooks() {
 		add_filter( 'ep_intercept_remote_request', '__return_true', 9999 );
 		add_filter( 'ep_do_intercept_request', [ $this, 'filter__ep_do_intercept_request' ], 9999, 4 );
 		add_filter( 'jetpack_active_modules', [ $this, 'filter__jetpack_active_modules' ], 9999 );
+
+		//Filter jetpack widgets
+		add_filter( 'jetpack_widgets_to_include', [ $this, 'filter__jetpack_widgets_to_include' ], 10, 1 );
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-elasticsearch.php]:
```suggestion
		add_filter( 'jetpack_widgets_to_include', [ $this, 'filter__jetpack_widgets_to_include' ], 10 );
```

Can leave off the args count param if there is just 1.

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 3]
```diff
@@ -54,6 +54,9 @@ protected function setup_hooks() {
 		add_filter( 'ep_intercept_remote_request', '__return_true', 9999 );
 		add_filter( 'ep_do_intercept_request', [ $this, 'filter__ep_do_intercept_request' ], 9999, 4 );
 		add_filter( 'jetpack_active_modules', [ $this, 'filter__jetpack_active_modules' ], 9999 );
+
+		//Filter jetpack widgets
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-elasticsearch.php]:
```suggestion
		// Filter jetpack widgets
```

File: elasticsearch/includes/classes/class-elasticsearch.php (hunk 1 above)
Comment (nickdaugherty) [reply elasticsearch/includes/classes/class-elasticsearch.php]: Switched over to only disabling search in 2bd452fc

**Comment by chrean** [comment]
Tested on a sandbox, can't see the Jetpack search in the Widgets page. Although, I can't see it if I use `master`, either. I feel I am doing something wrong.

**Comment by nickdaugherty** [comment]
> Tested on a sandbox, can't see the Jetpack search in the Widgets page

It's possible that JP Search wasn't enabled on the site you were testing - the module itself needs to first be enabled, then you need to add a JP Search widget to one of the sidebars on the site - then when you enable JP Search, the widgets should be disabled and the frontend should no longer use JP Search for searches.

**Comment by nickdaugherty** [comment]
I did some additional testing and everything looks good, going to merge + deploy.

**Comment by nickdaugherty** [comment]
r766-stacks

**Pull request merged**